stackslib/src/config/mod.rs

```diff
@@ -1656,52 +1656,220 @@ impl BurnchainConfigFile {
 
 #[derive(Clone, Debug)]
 pub struct NodeConfig {
+    /// Human-readable name for the node. Primarily used for identification in testing environments
+    /// (e.g., deriving log file names, temporary directory names).
+    ///
+    /// Default: `"helium-node"`
     pub name: String,
+    /// The node's Bitcoin wallet private key, provided as a hex string in the config file.
+    /// Used to initialize the node's keychain for signing operations.
+    /// If `miner.mining_key` is not set, this seed may also be used for mining-related signing.
+    /// Required if the node is configured as a miner (`node.miner = true`) and `miner.mining_key` is absent.
+    ///
+    /// Default: Randomly generated 32 bytes.
     pub seed: Vec<u8>,
+    /// The file system absolute path to the node's working directory.
+    /// All persistent data, including chainstate, burnchain databases, and potentially other stores,
+    /// will be located within this directory.
+    /// This path can be overridden by setting the `STACKS_WORKING_DIR` environment variable.
+    ///
+    /// Default: `/tmp/stacks-node-{current_timestamp}`.
     pub working_dir: String,
+    /// The IPv4 address and port (e.g., "0.0.0.0:20443") on which the node's HTTP RPC server
+    /// should bind and listen for incoming API requests.
+    ///
+    /// Default: `"0.0.0.0:20443"`
     pub rpc_bind: String,
+    /// The IPv4 address and port (e.g., "0.0.0.0:20444") on which the node's P2P networking
+    /// service should bind and listen for incoming connections from other peers.
+    ///
+    /// Default: `"0.0.0.0:20444"`
     pub p2p_bind: String,
+    /// The publicly accessible URL that this node advertises to peers during the P2P handshake
+    /// as its HTTP RPC endpoint. Other nodes or services might use this URL to query the node's API.
+    ///
+    /// Default: `http://{rpc_bind}` (e.g., "http://127.0.0.1:20443" if rpc_bind is default and resolves locally).
```

Comment:
if `rpc_bind` is defaulting to 0.0.0.0:20443 (above), wouldn't the default here be `http://0.0.0.0:20443`?